src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -184,138 +198,131 @@ export default function OpeningPeriodForm({
+              />
+              <p className="dash-between-begin-and-end-date">—</p>
+              <Datepicker
+                id="openingPeriodEndDate"
+                dataTest="openingPeriodEndDate"
+                labelText="Päättyy"
+                onChange={(value): void => setPeriodEndDate(value || null)}
+                value={periodEndDate}
+                registerFn={register}
+                customValidations={{
+                  dateRange: validateEndInputWithStartDate(periodBeginDate),
                 }}
-                iconLeft={<IconTrash />}>
-                Poista aukioloryhmä
-              </SupplementaryButton>
-            </div>
-            <input
-              type="hidden"
-              name={`${timeSpanGroupFieldName}[${index}].id`}
-              defaultValue={`${timeSpanGroup.id || ''}`}
-              ref={register()}
-            />
-            <input
-              type="hidden"
-              name={`${timeSpanGroupFieldName}[${index}].period`}
-              defaultValue={`${timeSpanGroup.period || ''}`}
-              ref={register()}
-            />
-            <TimeSpans
-              groupIndex={index}
-              groupId={timeSpanGroup.id}
-              namePrefix={timeSpanGroupFieldName}
-              control={control}
-              register={register}
-              resourceStateConfig={resourceStateConfig}
-            />
-            <Rules
-              groupIndex={index}
-              groupId={timeSpanGroup.id}
-              namePrefix={timeSpanGroupFieldName}
-              control={control}
-              register={register}
-              setValue={setValue}
-              ruleConfig={ruleConfig}
-            />
-          </section>
-        )
-      )}
-      <div className="form-actions-row form-actions-row-condensed">
-        <SupplementaryButton
-          dataTest="add-time-span-group"
-          onClick={(): void => appendTimeSpanGroup(defaultTimeSpanGroup)}
-          iconLeft={<IconPlus />}>
-          Luo uusi aukioloryhmä tähän jaksoon
-        </SupplementaryButton>
-        <p className="opening-period-notification-text">
-          <IconAlertCircle />
-          <span>
-            Lisää uusi ryhmä tähän aukiolojaksoon jos haluat lisätä
-            aukioloaikoja useammilla eri säännöillä
-          </span>
-        </p>
-      </div>
-      <div className="opening-period-final-action-row-container">
-        <PrimaryButton
-          dataTest="publish-opening-period-button"
-          className="opening-period-final-action-button"
-          type="submit">
-          Julkaise
-        </PrimaryButton>
-        <SecondaryButton
-          className="opening-period-final-action-button"
-          onClick={(): void => returnToResourcePage()}>
-          Peruuta ja palaa
-        </SecondaryButton>
-      </div>
-    </form>
+              />
+            </section>
+            {errors.openingPeriodEndDate?.type === 'dateRange' &&
```

Comment:
Are beginDate errors checked also?

Reply:
No, because we decided to go with the end date validation 🤔 
It has changes because it uses common ErrorText-component